services/services.go

```diff
@@ -109,111 +125,52 @@ type Advertisement interface {
 	SetEndpoint(e string) Advertisement          // sets the endpoint exposed by the service
 }
 
-// NewAdvertisement creates a new Advertisement
-func NewAdvertisement() Advertisement { return new(advertisement) }
+// ServiceQuery contains the fields required for service discovery queries
+type ServiceQuery interface {
+	Service() string                        // the service name of the query
+	SetService(s string) ServiceQuery       // set the service name of the query
+	Environment() string                    // the environemnt of the query
+	SetEnvironment(env string) ServiceQuery // sets the environemnt of the query
+	Zone() string                           // the zone of the query
+	SetZone(zone string) ServiceQuery       // sets the zone of the query
+}
 
 // QueryOptions are options to service discovery queries
 type QueryOptions interface {
-	Zones() []string                         // list of zones to consult. if empty only the local zone will be queried
-	SetZones(zones []string) QueryOptions    // sets the list of zones to consult
 	IncludeUnhealthy() bool                  // if true, will return unhealthy instances
 	SetIncludeUnhealthy(h bool) QueryOptions // sets whether to include unhealthy instances
 }
 
-// NewQueryOptions creates new QueryOptions
-func NewQueryOptions() QueryOptions { return new(queryOptions) }
-
 // Services provides access to the service topology
 type Services interface {
```

Comment:
Let's add a method as discussed that returns a `PlacementService` for a given `query ServiceQuery` then each parameter on `PlacementService` already has the context of a service.